sunpy/io/ana.py

```diff
@@ -50,6 +51,10 @@ def read(filename, debug=False, **kwargs):
     if _pyana is None:
         raise ImportError("C extension for ANA is missing, please rebuild.")
 
+    # NOTE: This can be removed after adding support for file-obj in `sunpy.io._pyana`.
+    if isinstance(filename, io.IOBase):
+        filename = filename.name  # Extracting path from the file-obj
```

Comment:
```suggestion
        filename = filename.name
```

Reply:
Is this tested in some form?

Reply:
Not currently, should it be?

Reply:
It might be tricky with ana but I am not pro-adding this without making sure it works in some form.